src/Makefile

```diff
@@ -116,24 +116,29 @@ endif
 
 ifeq ($(JULIACODEGEN),LLVM)
 ifneq ($(USE_SYSTEM_LLVM),0)
+# USE_SYSTEM_LLVM != 0
 CG_LLVMLINK += $(LLVM_LDFLAGS) $(shell $(LLVM_CONFIG_HOST) --libs --system-libs)
 # HACK: llvm-config doesn't correctly point to shared libs on all platforms
 #       https://github.com/JuliaLang/julia/issues/29981
 else
+# USE_SYSTEM_LLVM == 0
 ifneq ($(USE_LLVM_SHLIB),1)
+# USE_LLVM_SHLIB != 1
 CG_LLVMLINK += $(LLVM_LDFLAGS) $(shell $(LLVM_CONFIG_HOST) --libs $(CG_LLVM_LIBS) --link-static) $($(LLVM_LDFLAGS) $(shell $(LLVM_CONFIG_HOST) --system-libs 2> /dev/null)
 else
+# USE_LLVM_SHLIB == 1
 ifeq ($(OS), Darwin)
 CG_LLVMLINK += $(LLVM_LDFLAGS) -lLLVM
```

Comment:
Can we merge it with this branch as well?

Reply:
https://github.com/JuliaLang/julia/blob/05eb1530bbfe2db0dd6c9a13ea065f17cb9730cf/stdlib/libLLVM_jll/src/libLLVM_jll.jl#L23-L24
```
julia-mac/usr/lib/libLLVM-14.0.dylib -> libLLVM.dylib
```
Use `-lLLVM-14.0` or `-lLLVM`.

So at the moment we can't merge this branch.